pkg/controllers/monitoring-stack/components.go

```diff
@@ -485,10 +486,17 @@ func newPrometheus(
 	}
 
 	if ms.Spec.PrometheusConfig != nil {
-		prometheus.Spec.Storage = &monv1.StorageSpec{
-			VolumeClaimTemplate: monv1.EmbeddedPersistentVolumeClaim{
-				Spec: ms.Spec.PrometheusConfig.PersistentVolumeClaim,
-			},
+		// ms.Spec.PrometheusConfig.PersistentVolumeClaim return corev1.PersistentVolumeClaimSpec struct
+		// struct can not compare with nil so it checks if there is any thing set in the returned PersistentVolumeClaimSpec.
+		if !reflect.DeepEqual(ms.Spec.PrometheusConfig.PersistentVolumeClaim, corev1.PersistentVolumeClaimSpec{}) {
+			prometheus.Spec.Storage = &monv1.StorageSpec{
+				VolumeClaimTemplate: monv1.EmbeddedPersistentVolumeClaim{
+					Spec: ms.Spec.PrometheusConfig.PersistentVolumeClaim,
+				},
+			}
+		}
+		if ms.Spec.PrometheusConfig.RemoteWrite != nil {
```

Comment:
is this check needed? if not 

my suggestion is to create an empty `PrometheusConfig` and use that to initialise prometheus. This avoids the need for nil checks completely. 

```go
if ms.Spec.PromethesConfig == nil {
  ms.Spec.PrometheusConfig = defaultPrometheusConfig() // or PrometheusConfig{}
}

prometheus := monv1.Prometheus {
  Spec: monv1.PromtheusSpec {
     Storage: storageForConfig(ms.Spec.PrometheusConfig)
     RemoteWrite = remoteWriteFromConfig() // or  even ms.Spec.PrometheusConfig.RemoteWrite, nil is fine
  }
}
```